### PR TITLE
Upgrade panoptes-client

### PR DESCRIPTION
**Pull request by simoneduca**
Todos

- [x] Fix deployment (see screenshot)
screen shot 2018-01-31 at 14 45 02

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Is that the Docker build and deploy that's failing?

File: package.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
     "nib": "^1.1.0",
     "node-dir": "0.1.16",
     "npm": "^5.3.0",
-    "panoptes-client": "^2.2.0",
+    "panoptes-client": "^2.9.0",
```
Comment (eatyourgreens) [inline package.json]: Should this be "^2.9" so that patches to 2.9 are installed automatically?

**Comment by simoneduca** [comment]
AFA I can see, it's the gulp`browserify` task in the build that's broken. For ref https://github.com/terinjokes/gulp-uglify/issues/249#issuecomment-262406528

**Comment by simoneduca** [comment]
I thought I had seen this before https://github.com/zooniverse/shakespeares_world/pull/349#pullrequestreview-78874986

**Comment by eatyourgreens** [comment]
`npm run build` runs without error for me, locally. :man_shrugging:

**Comment by eatyourgreens** [comment]
The Panoptes client went through a pretty big jump here, 2.2 -> 2.9. Did one of those version changes introduce JavaScript that the gulp task can't parse? I have no idea.

**Comment by simoneduca** [comment]
But it fails also without the client upgrade. The mystery is that yesterday I published some changes with no problem.

**Comment by camallen** [comment]
It builds ok for on v8.9.3 for me.

**Comment by simoneduca** [comment]
If I use [pump like I had suggested](https://github.com/zooniverse/shakespeares_world/pull/349#issuecomment-346812574), it build fine in node 9.2

**Comment by simoneduca** [comment]
https://github.com/terinjokes/gulp-uglify/tree/master/docs/why-use-pump

**Comment by eatyourgreens** [comment]
Node 8.8 fails for me with the same error.

Does pump fix it, or just swallow the error and build a broken JS file?

**Comment by simoneduca** [comment]
`Pump` is supposed to allow any error to propagate, unlike `pipe`. So it should do the opposite of swallowing. We can test with staging the branch though.

**Comment by simoneduca** [comment]
Here's the pump PR https://github.com/zooniverse/shakespeares_world/pull/368

**Comment by eatyourgreens** [comment]
Fresh install, with node 9, builds for me without errors. I'll try this branch and report back.

**Comment by eatyourgreens** [comment]
`master` builds without problems. This branch fails. I don't know much about gulp or browserify, so I'm afraid I can't be much help here.

**Comment by camallen** [comment]
confirming master just built for me on v9.4, after a `git pull` && `npm install`. though `npm install` did update the `package-lock.json` as well.

**Comment by simoneduca** [comment]
I don't know what was going on, but after a fresh reinstall of the project, I've managed to build and publish this branch. Mind having a look? https://preview.zooniverse.org/shakespearesworld/#!/

**Comment by eatyourgreens** [comment]
Did it build 'js/main.js` or is preview using the JS file that's there from the last successful build?

**Comment by simoneduca** [comment]
I wasn't sure. So I did again and, yes, it did build it. Btw this isn't the branch using pump.

**Comment by eatyourgreens** [comment]
👍🏼 I can have a look tomorrow on a browser that has 3rd party cookies disabled.

**Comment by simoneduca** [comment]
Thanks! I tested disabling 3rd party cookies on Chrome and login works fine here. I was surprised the login persisted after ending the session though (closing tab). I thought session storage is not persisted in that case.

**Comment by eatyourgreens** [comment]
https://preview.zooniverse.org/shakespearesworld/js/main.js is an old version of the JS file, without the new client.

I still get `GulpUglifyError: unable to minify JavaScript` when I try to build this branch locally, after deleting `node_modules` and running a fresh `npm install`.

**Comment by eatyourgreens** [comment]
@simoneduca when I run the build, or the build and deploy tasks, the `build/` directory isn't cleaned up after they run, even if the build failed. Are you, perhaps, deploying an old JS file from your machine, even though the build is actually failing?

I've also noticed that the build and deploy completes without error even if `js/main.js` isn't actually built.

**Comment by simoneduca** [comment]
@eatyourgreens I can see `"panoptes-client":"^2.9.0"` in the link you pasted above. Isn't that latest version? My `build` dir is cleaned every time. This is weird...

**Comment by simoneduca** [comment]
Could you do me a favour and try cloning a fresh copy of the project?

**Comment by eatyourgreens** [comment]
The client bundled in `js/main.js` on preview isn't 2.9. I checked this by searching through the source map for the oauth code in the Chrome dev console.

`npm run build` works with version 2.8.1 of the client, but not 2.9. I can't see what is tripping up uglify though.

**Comment by eatyourgreens** [comment]
I am working with a completely fresh copy of SW.

**Comment by eatyourgreens** [comment]
[Adding ES6 support to uglify](https://www.npmjs.com/package/gulp-uglify#using-a-different-uglifyjs) fixes the build.

**Comment by eatyourgreens** [comment]
For future reference, I had a completely clean copy of SW. Between builds, I ran this to check that the JS file was being built:
```
rm -rf build
rm -rf node_modules
npm install
npm run build
```

**Comment by simoneduca** [comment]
Ack! Sounds obvious now, thanks! I'll stage the brach.

**Comment by eatyourgreens** [comment]
> @eatyourgreens I can see "panoptes-client":"^2.9.0" in the link you pasted above. Isn't that latest version? My build dir is cleaned every time. This is weird...

I think you might have been looking at a copy that I had deployed.

**Comment by eatyourgreens** [comment]
The Uglify error was "Name expected", but I can't see what changed in the client JS that would break uglify. However, the version of uglify that SW was using can't process ES6 files.

**Comment by simoneduca** [comment]
Maybe, but I was looking at the preview before you push the es6 support code. Does it need staging now?

File: package.json (hunk 1 above)
Comment (simoneduca) [reply package.json]: I'm not sure. PFE doesn't do that.

**Comment by eatyourgreens** [comment]
I've just run `npm run deploy-staging` again, from this branch. It would be handy if the deploy reported which files it was uploading and overwriting, like `publisssh` does. Is that something you can change?

**Comment by simoneduca** [comment]
I can certainly try, but in a different PR perhaps? I've tested the staged branch and it looks fine. Is this ready to go then?

**Comment by simoneduca** [comment]
This line in client is breaking `uglify` https://github.com/zooniverse/panoptes-javascript-client/blob/master/lib/oauth.js#L281 Adding ES6 support would let the code through, but older browsers would break. Thanks @eatyourgreens!
So we're changing that line on the client, publish the patch and upgrading SW here.

**Comment by eatyourgreens** [comment]
I'm seeing this now, when I deploy to staging, which I wasn't seeing before so that's good.
```
[14:13:39] Uploading files to zooniverse-static/preview.zooniverse.org/shakespearesworld/...
[14:13:59] Uploaded build/js/main.js.gz
[14:14:09] Uploaded build/js/main.js
[14:14:28] Uploaded build/js/main.js.map.gz
```

**Comment by eatyourgreens** [comment]
@simoneduca your link doesn't link to the broken code, now that it's been fixed.

**Comment by simoneduca** [comment]
I'll fix the link, but I'm getting a fresh install because I was still seeing the same error after pulling the latest on this brach.

**Comment by simoneduca** [comment]
Looks good now. I can see the correct client version in `main.js` and works fine on preview. Merging.

**Comment by simoneduca** [comment]
I'm also deploying manually to prod, so we can test.

**Pull request merged**